.github/workflows/golden_tests.yml

```
@@ -0,0 +1,101 @@
+      
+      # Run tests to check for differences
+      - name: Run golden tests
+        id: golden-test-run
+        run: |
+          echo "Running golden tests to identify UI differences..."
+          flutter test --tags golden || echo "::set-output name=has_differences::true"
+      
+      # Save golden files and failure images as artifacts
+      - name: Upload golden test artifacts
+        if: always()
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)